crt_portal/cts_forms/forms.py

```diff
@@ -2028,6 +2028,167 @@ class PrintActions(Form):
+                },
+            ),
+            required=True,
+            disabled=self.instance.first_reviewer is not None
+        )
+
+    def setup_first_review_date(self):
+        first_review_date = self.instance.first_review_date if self.instance.first_review_date else datetime.today()
+        self.fields['first_review_date'] = CharField(
+            required=True,
+            label="Date",
+            widget=CrtTextInput(attrs={
+                'class': 'usa-input',
+                'name': 'first_review_date',
+                'placeholder': 'mm/dd/yyyy',
+                'aria_label': 'Date',
+                'value': first_review_date.strftime('%m/%d/%Y'),
+                'label': 'Date',
+            }),
+            disabled=self.instance.first_review_date is not None
+        )
+
+    def setup_second_reviewer(self):
+        second_reviewer = self.instance.second_reviewer if self.instance.second_reviewer else self.user
+        value = f'{second_reviewer.first_name} {second_reviewer.last_name}' if second_reviewer.first_name and second_reviewer.last_name else ''
+        self.fields['second_reviewer'] = CharField(
+            label='Secondary Authorizing Individual',
+            widget=CrtTextInput(
+                attrs={
+                    'class': 'usa-input',
+                    'name': 'second_reviewer',
+                    'placeholder': 'Secondary Authorizing Individual',
+                    'aria_label': 'Secondary Authorizing Individual',
+                    'value': value,
+                    'label': 'Secondary Authorizing Individual',
+                },
+            ),
+            required=self.instance.first_reviewer is not None,
+            disabled=self.instance.second_reviewer is not None,
+        )
+
+    def setup_second_review_date(self):
+        second_review_date = self.instance.second_review_date if self.instance.second_review_date else datetime.today()
+        display_value = second_review_date.strftime('%m/%d/%Y')
+        self.fields['second_review_date'] = CharField(
+            required=self.instance.first_review_date is not None,
+            label="Date",
+            widget=CrtTextInput(attrs={
+                'class': 'usa-input',
+                'name': 'second_review_date',
+                'placeholder': 'mm/dd/yyyy',
+                'aria_label': 'Date',
+                'value': display_value,
+                'label': 'Date',
+            }),
+            disabled=self.instance.second_review_date is not None
+        )
+
+    def clean_first_reviewer(self):
+        user = self.user.get_username()
+        if 'first_reviewer' not in self.cleaned_data or self.instance.first_reviewer is not None:
+            return self.instance.first_reviewer
+        name = self.cleaned_data['first_reviewer'].split(' ')
+        if len(name) == 2:
+            user = User.objects.filter(first_name=name[0], last_name=name[1]).first()
+        return user
```

Comment:
oh interesting - when we submit this form, we can't have the display be the user's name but the value be the user's id?

for example, there's definitely two amanda sanchez at DOJ

Reply:
Good catch! Updated this to use the user id